lmnet/lmnet/datasets/dataset_iterator.py

```diff
@@ -208,8 +211,17 @@ def __init__(self, dataset, enable_prefetch=False, seed=0):
         self.seed = seed
 
         if self.enable_prefetch:
-            self.prefetch_result_queue = queue.Queue(maxsize=200)
-            self.prefetcher = _MultiProcessDatasetPrefetchThread(self.dataset, self.prefetch_result_queue, seed)
+            queue_size = self.dataset.batch_size
+            if self.dataset.num_parallel_process and self.dataset.num_parallel_process > 0:
```

Comment:
Please set num_process default value in DatasetItelator’s  __init__() .

Reply:
@ruimashita 
Set default value of num_parallel_process in init(). If set in dataset config it will use it or else set to half of the available cpu count. The config is not mandatory.

Reply:
@ananno 
Sorry I mistaken. 😭  The `num_process` default value is set in DatasetItelator’s init() .